### PR TITLE
Use AddComponentMenu for MonoBehaviours

### PR DESCRIPTION
**Pull request by chriselion**
There have been a few issues about missing menu items. 
https://github.com/Unity-Technologies/ml-agents/issues/2870
https://github.com/Unity-Technologies/ml-agents/issues/3227

These were probably from following a now-outdated tutorial, but I think adding menu items for the MonoBehaviours that we have probably makes things a bit more discoverable:

Menu bar:
Screen Shot 2020-01-15 at 1 18 11 PM

Add Component in the inspector
Screen Shot 2020-01-15 at 1 21 11 PM

Unfortunately, `Agent` doesn't qualify for this because it's abstract.

### CODE REVIEW

File: UnitySDK/Assets/ML-Agents/Scripts/DemonstrationRecorder.cs [hunk 1]
```diff
@@ -10,6 +10,7 @@ namespace MLAgents
     /// Demonstration Recorder Component.
     /// </summary>
     [RequireComponent(typeof(Agent))]
+    [AddComponentMenu("ML Agents/Demonstration Recorder")]
```
Comment (chriselion) [inline UnitySDK/Assets/ML-Agents/Scripts/DemonstrationRecorder.cs]: Should we list this? I'd be willing to remove it.

File: UnitySDK/Assets/ML-Agents/Scripts/Monitor.cs [hunk 2]
```diff
@@ -8,6 +8,7 @@ namespace MLAgents
     /// Monitor is used to display information about the Agent within the Unity
     /// scene. Use the log function to add information to your monitor.
     /// </summary>
+    [AddComponentMenu("ML Agents/Monitor")]
```
Comment (chriselion) [inline UnitySDK/Assets/ML-Agents/Scripts/Monitor.cs]: I'm OK removing this one too.

File: UnitySDK/Assets/ML-Agents/Scripts/Sensor/CameraSensorComponent.cs [hunk 3]
```diff
@@ -3,6 +3,7 @@
 
 namespace MLAgents.Sensor
 {
+    [AddComponentMenu("ML Agents/Sensors/Camera Sensor")]
```
Comment (chriselion) [inline UnitySDK/Assets/ML-Agents/Scripts/Sensor/CameraSensorComponent.cs]: Intentionally left "Component" off the sensor menu items; I think it's a bit wordy otherwise but willing to change it.

**Comment by vincentpierre** [comment]
> Unfortunately, Agent doesn't qualify for this because it's abstract.

For now 😈

File: UnitySDK/Assets/ML-Agents/Scripts/Monitor.cs (hunk 2 above)
Comment (vincentpierre) [reply UnitySDK/Assets/ML-Agents/Scripts/Monitor.cs]: We should, this is supposed to be hidden from the user and automagic

File: UnitySDK/Assets/ML-Agents/Scripts/DemonstrationRecorder.cs (hunk 1 above)
Comment (vincentpierre) [reply UnitySDK/Assets/ML-Agents/Scripts/DemonstrationRecorder.cs]: I think it makes sense

Comment (chriselion) [reply UnitySDK/Assets/ML-Agents/Scripts/DemonstrationRecorder.cs]: Makes sense to have it or remove it?

File: UnitySDK/Assets/ML-Agents/Scripts/Sensor/CameraSensorComponent.cs (hunk 3 above)
Comment (surfnerd) [reply UnitySDK/Assets/ML-Agents/Scripts/Sensor/CameraSensorComponent.cs]: The Unity UI guidelines recommend against nesting menus.  We should just group the alike menu items using the `componentOrder` constructor variable.

Comment (surfnerd) [reply UnitySDK/Assets/ML-Agents/Scripts/Sensor/CameraSensorComponent.cs]:
```suggestion
    [AddComponentMenu("ML Agents/Camera Sensor", kSensorMenuGroup)]
```
Where `kSensorMenuGroup` is defined somewhere.

**Comment by chriselion** [comment]
Without nesting and adding a menu group:
Screen Shot 2020-01-16 at 3 08 35 PM

File: UnitySDK/Assets/ML-Agents/Scripts/Constants.cs [hunk 4]
```diff
@@ -0,0 +1,11 @@
+namespace MLAgents
+{
+    /// <summary>
+    /// Grouping for use in AddComponentMenu (instead of nesting the menus).
+    /// </summary>
+    public enum MenuGroup
```
Comment (chriselion) [inline UnitySDK/Assets/ML-Agents/Scripts/Constants.cs]: Better way to do this / better place to put it?

Comment (surfnerd) [reply UnitySDK/Assets/ML-Agents/Scripts/Constants.cs]: Maybe `EditorConstants` and put it in the Editor folder.
